src/dirbrowserformaction.cpp

```diff
@@ -334,13 +334,13 @@ void DirBrowserFormAction::add_directory(
 	}
 }
 
-Filepath DirBrowserFormAction::get_formatted_dirname(const Filepath& dirname,
+std::string DirBrowserFormAction::get_formatted_dirname(const Filepath& dirname,
 	mode_t mode)
 {
-	Filepath tmp = dirname;
+	std::string tmp = dirname.to_locale_string();
 	const auto suffix = file_system::mode_suffix(mode);
 	if (suffix.has_value()) {
-		tmp.set_extension(std::to_string(suffix.value()));
```

Comment:
I'm wondering how this even worked, especially given that `suffix` would be `/` in case of directories.
[PathBuf's `set_extension()`](https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.set_extension) seems to panic in case of the string containing a path separator character.

Reply:
Oh, looks like `std::to_string()` on a char casts the char to an int and returns the numeric value.
I expect the line as it is now requires an update to get back an actual single-character string.

I probably missed that during one of my previous 2326-related PRs.

Reply:
I expect that the FileBrowser variant (`strprintf::fmt("%s%c", filename_str, suffix.value());`) works fine, so that could be applied here as well.

Reply:
Right! However, the existing `strprintf::fmt` code was also buggy, because `filename_str` was a string in locale encoding; `strprintf::fmt` should work on UTF-8 strings. I fixed al that in 44853874d76e42e3a8a11d501a6c610870023c5e.